### PR TITLE
IDEX-2855: Remove extensions servers info from non dev machines

### PR DESCRIPTION
**Pull request by garagatyi**
@skabashnyuk @evoevodin

### CODE REVIEW

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java [hunk 1]
```diff
@@ -259,18 +275,39 @@ private Instance createInstance(String imageId,
...
                                     Recipe recipe,
                                     int memorySizeMB,
                                     LineConsumer outputConsumer)
             throws MachineException {
         try {
+            final Map<String, String> labels;
+            final Map<String, Map<String, String>> portsToExpose;
+            final Set<String> volumes;
+            if (isDev) {
+                labels = new HashMap<>(machineContainerLabels.size() + devMachineContainerLabels.size());
```
Comment (evoevodin) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: What do you think about <code>Maps.newHashMapWithExpectedSize(...)</code> in this case?

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java [hunk 2]
```diff
@@ -259,18 +275,39 @@ private Instance createInstance(String imageId,
...
+                labels.putAll(machineContainerLabels);
+                labels.putAll(devMachineContainerLabels);
+
+                portsToExpose = new HashMap<>(portsToExposeOnMachine.size() + portsToExposeOnDevMachine.size());
+                portsToExpose.putAll(portsToExposeOnMachine);
+                portsToExpose.putAll(portsToExposeOnDevMachine);
+
+                // 1 extra element that contains workspace FS folder will be added further
+                volumes = new HashSet<>(devMachineSystemVolumes.size() + systemVolumesForMachine.size() + 1);
```
Comment (evoevodin) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: What about <code>Sets.newHashSetWithExpectedSize(...)</code> in this case?

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java (hunk 1 above)
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: +1 10x

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java (hunk 2 above)
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: +1

**Comment by skabashnyuk** [comment]
LGTM

**Pull request closed without merging**